swagger_server/itm/itm_action_handler.py

```diff
@@ -0,0 +1,416 @@
+import json
+from swagger_server.models import (
+    Action,
+    Casualty
+)
+from .itm_session_scenario_object import ITMSessionScenarioObject
+
+class ITMActionHandler:
+    """
+    Class for validating and processing actions.
+    """
+
+    def __init__(self, session):
+        """
+        Initialize an ITMActionHandler.
+        """
+        from.itm_scenario_session import ITMScenarioSession
+        self.session: ITMScenarioSession = session
+        self.current_isso: ITMSessionScenarioObject = None
+        with open("swagger_server/itm/treatment_times_config/actionTimes.json", 'r') as json_file:
+                self.times_dict = json.load(json_file)
+
+    def set_isso(self, isso):
+        self.current_isso = isso
+
+
+    def _reveal_injuries(self, source: Casualty, target: Casualty):
+        if target.visited: # Don't reveal injuries in visited casualties
+            return
+
+        for source_injury in source.injuries:
```

Comment:
If you are interested in not having a nested for loop here we could split this into a separate function like this.

```python
def should_reveal_injury(source_injury, target):
        return source_injury.name in hidden_injury_types and \
               not any(target_injury.name in hidden_injury_types for target_injury in target.injuries)
```

And then call it inside _reveal_injuries like this:

```python
revealed_injuries = [source_injury for source_injury in source.injuries if should_reveal_injury(source_injury, target)]
target.injuries.extend(revealed_injuries)
```

Reply:
I think ultimately the "hiddenness" of injuries will be a property of the injury and not the scenario, so this code will need to be revisited.  I made your change in the meantime.